packages/rn-tester/js/examples/ActivityIndicator/ActivityIndicatorExample.js

```diff
@@ -10,46 +10,35 @@
 
 import type {Node} from 'React';
 import {ActivityIndicator, StyleSheet, View} from 'react-native';
-import React, {Component} from 'react';
+import React, {useState, useCallback, useRef, useEffect} from 'react';
 
-type State = {|animating: boolean|};
-type Props = $ReadOnly<{||}>;
-type Timer = TimeoutID;
+function ToggleAnimatingActivityIndicator() {
+  let _timer = useRef();
 
-class ToggleAnimatingActivityIndicator extends Component<Props, State> {
-  _timer: Timer;
+  const [animating, setAnimating] = useState(true);
 
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      animating: true,
-    };
-  }
-
-  componentDidMount() {
-    this.setToggleTimeout();
-  }
+  const setToggleTimeout = useCallback(() => {
+    _timer.current = setTimeout(() => {
+      setAnimating(currentState => !currentState);
+      setToggleTimeout();
+    }, 2000);
+  }, []);
```

Comment:
```suggestion
  }, [animating]);
```

Reply:
It's not necessary given I get the current state from the setter, right?

Reply:
Yeah I think we are fine here since `setAnimating` is using the callback form of `setState`, and the function should not have to change depending on value.